packages/core/admin/admin/src/pages/MarketplacePage/index.js

```diff
@@ -16,96 +13,41 @@ import { Layout, ContentLayout } from '@strapi/design-system/Layout';
 import { Main } from '@strapi/design-system/Main';
 import { Searchbar } from '@strapi/design-system/Searchbar';
 import { Box } from '@strapi/design-system/Box';
-import { useNotifyAT } from '@strapi/design-system/LiveRegions';
-import { Typography } from '@strapi/design-system/Typography';
 import { Flex } from '@strapi/design-system/Flex';
 import { Tabs, Tab, TabGroup, TabPanels, TabPanel } from '@strapi/design-system/Tabs';
 
-import EmptyNpmPackageSearch from './components/EmptyNpmPackageSearch';
 import PageHeader from './components/PageHeader';
-import useFetchMarketplaceProviders from '../../hooks/useFetchMarketplaceProviders';
-import useFetchMarketplacePlugins from '../../hooks/useFetchMarketplacePlugins';
 import adminPermissions from '../../permissions';
-import offlineCloud from '../../assets/images/icon_offline-cloud.svg';
 import useNavigatorOnLine from '../../hooks/useNavigatorOnLine';
 import MissingPluginBanner from './components/MissingPluginBanner';
 import NpmPackagesGrid from './components/NpmPackagesGrid';
 import SortSelect from './components/SortSelect';
 import NpmPackagesFilters from './components/NpmPackagesFilters';
 import NpmPackagesPagination from './components/NpmPackagesPagination';
-
-const matchSearch = (npmPackages, search) => {
-  return matchSorter(npmPackages, search, {
-    keys: [
-      {
-        threshold: matchSorter.rankings.WORD_STARTS_WITH,
-        key: 'attributes.name',
-      },
-      { threshold: matchSorter.rankings.WORD_STARTS_WITH, key: 'attributes.description' },
-    ],
-    baseSort: (a, b) => (a.index < b.index ? -1 : 1),
-  });
-};
+import useDebounce from '../../content-manager/components/InputUID/useDebounce';
```

Comment:
I agree that using the debounce from InputUID is a bit weird, I'll see if it's possible to move it up.

Regarding the onSubmit, I discussed it with Maeva, and we thought keeping the search-as-you-type would be nicer. It's how the marketplace has worked so far, so since there's no "enter" button, users may be confused if what worked before stops working

Regarding the content manager search, it's a different component that's used there (`SearchURLQuery` vs `SearchBar`). That SearchURLQuery component is a bit different as it's meant to open and collapse and it's smaller. So I think if we wanted the search on submit we'd add that functionality to SearchBar instead